File: leetcode code/16-3sum-closest/16-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums);
        int result = nums[0]+nums[1]+nums[2];
        
        for(int i=0;i<nums.length;i++){
            int left=i+1;
            int right=nums.length-1;
            while(left<right){
                int dummy = nums[i] + nums[left] + nums[right];
                if(dummy==target){
                   return dummy; 
                }
                if(Math.abs(dummy-target)<Math.abs(result-target)){
                    result = dummy;
                }
                if(dummy<target){
                    left += 1;
                }
                else{
                    right -= 1;
                }
            }
        }
        return result;
    }
}

File: leetcode code/16-3sum-closest/16-3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        temp = nums[0] + nums[1] + nums[2]
        for i in range(len(nums)-2):
            left,right = i+1,len(nums)-1
            while left<right:
                dummy = nums[i] + nums[left] + nums[right]
                if dummy == target:
                    return dummy
                if abs(dummy-target) < abs(temp-target):
                    temp = dummy
                if dummy<target:
                    left+=1
                else:
                    right-=1
        return temp
                